pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -121,28 +121,55 @@ func (r *ReconcileUserAccount) Reconcile(request reconcile.Request) (reconcile.R
 	// If the UserAccount has not been deleted, create or update user and identity resources.
 	// If the UserAccount has been deleted, delete secondary resources identity and user.
 	if !util.IsBeingDeleted(userAcc) {
+		reqLogger.Info("Adding finalizer on UserAccount")
 		// Add the finalizer if it is not present
 		if err := r.addFinalizer(userAcc); err != nil {
 			return reconcile.Result{}, err
 		}
-
+	}
+	if !util.IsBeingDeleted(userAcc) && !userAcc.Spec.Disabled {
+		reqLogger.Info("Ensuring user and identity associated with UserAccount")
 		var createdOrUpdated bool
 		var user *userv1.User
 		if user, createdOrUpdated, err = r.ensureUser(reqLogger, userAcc); err != nil || createdOrUpdated {
 			return reconcile.Result{}, err
 		}
-
 		if _, createdOrUpdated, err = r.ensureIdentity(reqLogger, userAcc, user); err != nil || createdOrUpdated {
 			return reconcile.Result{}, err
 		}
-
 		if _, createdOrUpdated, err = r.ensureNSTemplateSet(reqLogger, userAcc); err != nil || createdOrUpdated {
 			return reconcile.Result{}, err
 		}
-	} else if util.HasFinalizer(userAcc, userAccFinalizerName) {
-		if err = r.manageCleanUp(userAcc); err != nil {
+	} else if util.HasFinalizer(userAcc, userAccFinalizerName) && util.IsBeingDeleted(userAcc) {
+		reqLogger.Info("Deleting user and identity associated with UserAccount")
+		deleted, err := r.deleteUserAndIdentity(userAcc)
+		if err != nil {
 			return reconcile.Result{}, err
```

Comment:
Should there be a status update here?

Reply:
hmm, i'm not too sure, what would we set the status update to? i figured we wouldn't need to update the status here because something went wrong - we either couldn't get the user/identity (for reasons other than "is not found") or the deletion failed. In which case, i'd like to just retry because this has to be completed before the useraccount can be deleted..